File: core/src/main/java/cz/dynawest/jtexy/parsers/TexyEventListener.java

package cz.dynawest.jtexy.parsers;

import cz.dynawest.jtexy.TexyException;
import cz.dynawest.jtexy.events.TexyEvent;
import org.dom4j.Node;

/**
 *
 * @author Ondrej Zizka
 */
public abstract interface TexyEventListener<T extends TexyEvent> {

	public Class getEventClass();

	public Node onEvent( T event ) throws TexyException;

}// class

File: core/src/main/java/cz/dynawest/jtexy/parsers/TexyEventListener.kt
package cz.dynawest.jtexy.parsers

import cz.dynawest.jtexy.TexyException
import cz.dynawest.jtexy.events.TexyEvent
import org.dom4j.Node

/**
 *
 * @author Ondrej Zizka
 */
interface TexyEventListener<in T : TexyEvent> {
    val eventClass: Class<*>

    @Throws(TexyException::class)
    fun onEvent(event: T): Node?
}
